tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_max_pool.py

```diff
@@ -322,7 +252,7 @@ def copy_input_to_device(self, input: ttl.tensor.Tensor):
         in_n = self.sliding_window_op_params.batch_size
         in_h = self.sliding_window_op_params.input_h
         in_w = self.sliding_window_op_params.input_w
-        assert in_c % 16 == 0, "Input channels should be multiple of 32. General case is TODO"
+        assert in_c % 32 == 0, "Input channels should be multiple of 32. General case is TODO"
```

Comment:
Why is this change needed? This will disable the shallow depth maxpool?

Reply:
Doesn't this mean that metal would have to support half tiles?  I didn't see the below code pad the input, unless this is automatically handled?

Reply:
Reverted it back! Still curious about my question though :)